crates/autopilot/src/infra/blockchain/dto.rs

```diff
@@ -1,24 +1,29 @@
-impl TryFrom<web3::types::Transaction> for crate::domain::settlement::Transaction {
-    type Error = &'static str;
+impl TryFrom<(web3::types::Transaction, web3::types::TransactionReceipt)>
+    for crate::domain::settlement::Transaction
+{
+    type Error = anyhow::Error;
 
-    fn try_from(value: web3::types::Transaction) -> Result<Self, Self::Error> {
+    fn try_from(
+        (transaction, receipt): (web3::types::Transaction, web3::types::TransactionReceipt),
```

Comment:
Try from looks very sketchy for tuples. Maybe it's time for a `try_new` constructor.

Reply:
How would that look like? I am pretty happy with the existing option tbh

Reply:
The issue with `try_from` is that it makes it hard to find the code that actually implements the logic. If I'm on 

https://github.com/cowprotocol/services/blob/d417d97a197af0f4a15372b5e69b08493c843feb/crates/autopilot/src/infra/blockchain/mod.rs#L120-L125

and click on `try_into`, I'm taken to stdlib impl of `try_into` and it's hard to find this actual code here. This would not happen if this was a custom method that isn't part of any trait.

Is there any reason to use the standard trait here rather than a free function other than "syntactic appeal" ?

Reply:
> How would that look like? I am pretty happy with the existing option tbh

Could be as simple as
```rust
fn recover_transaction(tx: web3::types::Transaction, receipt: web3::types::TransactionReceipt) -> Result<domain::Transaction> {}
```
Also writing it out like this kind of points out that this does not seem like a dto functionality to me.

Reply:
Moved out and created a free function `into_domain` for it.